app/decorators/case_transition_decorator.rb

```diff
@@ -19,18 +19,24 @@ def event_and_detail
 
   private
 
-  def event
-    state_machine = object.case.state_machine
-    state_machine.event_name(object.event)
+  def event_desc
+    description_for_event = has_detail_description_for_event
```

Comment:
You can get rid of this whole line if you like - we'd tend to reserve methods named things like "has_whatever_" for methods that return boolean values - we'd also name it with a trailing `?` in that case. Here I'd consider just calling the method `description_for_event` and losing the local variable altogether

Reply:
@Polsonby , same thing I thought I may had better use your suggested name lol :), let me change it